src/app/controllers/PokemonController.js

```diff
@@ -0,0 +1,72 @@
+import Pokemon from '../models/pokemon';
+import paginationHandler from '../services/paginationHandler';
+
+class PokemonController {
+  async getInfo(req, res) {
+    const info = [];
+    info.push({
+      description: 'Banco de dados com informações referentes ao jogo Pokemon Go!',
+      tableSize: await Pokemon.count(),
+      pokemonAttributes: await Pokemon.getAttributes(),
+    });
+
+    return res.json(info);
+  }
+
+  async listAll(req, res) {
+    const { limit, offset, page, totalPages } = await paginationHandler(req.query);
+    const pokemons = await Pokemon.findAndCountAll({ limit, offset });
+
+    const response = {
+      totalItens: pokemons.count,
+      paginationSize: limit,
+      totalPages,
+      currentPage: page,
+      pokemons: pokemons.rows,
+    };
+
+    return res.json(response);
+  }
+
+  async getById(req, res) {
+    const { id } = req.params;
+    const pokemon = await Pokemon.findByPk(id);
+    if (pokemon) {
+      return res.json(pokemon);
+    } else {
```

Comment:
uma dica, como está fazendo o retorno antecipado dentro do `if`, não precisa do bloco `else`, assim o código fica mais legível:
```suggestion
   if (pokemon) return res.json(pokemon);
   return res.status(404).json({ error: `The id ${id} does not exist in the database.` });
```
Funciona exatamente do mesmo jeito e ao mesmo tempo reduz a complexidade ciclomática.

Reply:
Entendi, vou fazer as alterações aqui, fica bem melhor mesmo.